### PR TITLE
QueryLogics can optionally pull accumulo passwords from the environment

### PR DESCRIPTION
**Pull request by apmoriarty**
Addresses https://github.com/NationalSecurityAgency/datawave/issues/1696

### CODE REVIEW

File: web-services/query/src/main/java/datawave/webservice/query/configuration/GenericQueryConfiguration.java [hunk 1]
```diff
@@ -167,6 +170,48 @@ public void setBypassAccumulo(boolean bypassAccumulo) {
...
+     *
+     * @param accumuloPasswordEnv
+     *            the environment variable where the accumulo password is stored
+     */
+    public void setAccumuloPasswordEnv(String accumuloPasswordEnv) {
+        this.accumuloPasswordEnv = accumuloPasswordEnv;
+        if (StringUtils.isNotBlank(accumuloPasswordEnv)) {
+            String password = System.getenv(accumuloPasswordEnv);
+            if (StringUtils.isNotBlank(password)) {
+                this.accumuloPassword = password;
```
Comment (avgAGB) [inline web-services/query/src/main/java/datawave/webservice/query/configuration/GenericQueryConfiguration.java]: splitting hairs, but would it make sense to use the setter for the accumulo password rather then directly modding it here?

Comment (apmoriarty) [reply web-services/query/src/main/java/datawave/webservice/query/configuration/GenericQueryConfiguration.java]: works for me. I'll add it in the next commit

**Review by avgAGB — DISMISSED** [review]

**Review by hlgp — DISMISSED** [review]

File: warehouse/query-core/src/main/java/datawave/query/metrics/AccumuloRecordWriter.java [hunk 2]
```diff
@@ -298,7 +299,8 @@ protected static String getUsername(Configuration conf) {
      * string, and is not intended to be secure.
      */
     protected static byte[] getPassword(Configuration conf) {
-        return Base64.decodeBase64(conf.get(PASSWORD, "").getBytes());
+        String pw = EnvProvider.resolve(PASSWORD);
```
Comment (FineAndDandy) [inline warehouse/query-core/src/main/java/datawave/query/metrics/AccumuloRecordWriter.java]: will this be base64 encoded?

**Review by FineAndDandy — CHANGES_REQUESTED** [review]

File: warehouse/query-core/src/main/java/datawave/query/metrics/AccumuloRecordWriter.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/query/metrics/AccumuloRecordWriter.java]: This whole code block was wrong, good catch.

**Review by ivakegg — APPROVED** [review]

**Review by FineAndDandy — APPROVED** [review]

**Pull request merged**